### PR TITLE
write out TPMinV to assets file only for csharp uwp projects

### LINKED ISSUE #5815: Upgrade to Visual Studio 15.3 regressed package.json build behaviour for C++ UWP projects
Please read the following information before posting the issue.
[Dll1.zip](https://github.com/NuGet/Home/files/1258498/Dll1.zip)

## Details about Problem

*NuGet product used:*
VS UI

*NuGet version (x.x.x.xxx):*
That which was shipped in VS 2017 Update 3

*VS version (if appropriate):*
26815.3 [and prior]

*OS version (i.e. win10 v1607 (14393.321)):*
win10 v1703 and insider builds

*Worked before?*
_Yes, with VS 2017 Update 2._

## Detailed repro steps so we can see the same problem

1. Create a project of type `DLL (Universal Windows)`

2. Add a project.json with no dependencies (or, any number of _native_ dependencies)

```json
{
  "dependencies": { },
  "frameworks": {
    "uap10.0": {
      "imports": [
        "native",
        "uap"
      ]
    }
  }
}
```

3. Restore Packages for Solution

4. Build All

5. The build fails:

```
1>------ Rebuild All started: Project: Dll1, Configuration: Debug Win32 ------
...
1>Target ResolveNuGetPackageAssets:
1>    C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\Microsoft\NuGet\15.0\Microsoft.NuGet.targets(178,5):
   error : Your project is not referencing the "UAP,Version=v10.0" framework.
   Add a reference to "UAP,Version=v10.0" in the "frameworks" section of your
   project.json, and then re-run NuGet restore.
1>Done building target "ResolveNuGetPackageAssets" in project "Dll1.vcxproj" -- FAILED.
1>
1>Done building project "Dll1.vcxproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

...

### Investigation

Restoring a native project with a project.json in Visual Studio¬, even if it doesn’t consume NuGet.Build.Packaging, results in the following:

Project.lock.json is populated with framework references:

```
{
  "version": 3,
  "targets": {
    "UAP,Version=v10.0.10586": {
      ...
    "UAP,Version=v10.0.10586/win10-arm": {
      ...
    "UAP,Version=v10.0.10586/win10-arm-aot": {
      ...
    "UAP,Version=v10.0.10586/win10-x86": {
      ...
    "UAP,Version=v10.0.10586/win10-x86-aot": {
      ...
```

However, the Preferred Target Moniker here is UAP,Version=v10.0.

Microsoft.NuGet.Build.Tasks' [ResolveNuGetPackageAssets.cs:660](https://github.com/NuGet/NuGet.BuildTasks/blob/dev/src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs#L660) uses that preferred target moniker to look up the target in project.lock.json. Clearly, that doesn’t exist.

It looks like either package restore is putting too much version information in project.lock.json or ResolveNuGetPackageAssets doesn’t support any manner of version fallback. It doesn’t strain belief that v10.0.10586 should provide v10.0.

This reproduces in 15.3’s public release and 15.4 preview.

### Sample Project

[Dll1.zip](https://github.com/NuGet/Home/files/1258499/Dll1.zip)

### Real-world example

We have a large set of projects consuming native dependencies via project.json at [WinObjC](http://github.com/Microsoft/WinObjC). As of VS15.3 the in-tree build for these projects fails with the above error.

### PR DESCRIPTION
**Pull request by rohit21agrawal**
fixes: https://github.com/NuGet/Home/issues/5815

The full context of what the regression is and why it was caused is explained in the bug: https://github.com/NuGet/Home/issues/5815#issuecomment-326055841

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectJsonNuGetProject.cs [hunk 1]
```diff
@@ -76,7 +76,15 @@ internal class VsProjectJsonNuGetProject : ProjectJsonNuGetProject
...
-                    if (platformMinVersion != null && jsonTargetFramework.Version != platformMinVersion)
+                    var targetFrameworkMonikerString = await _vsProjectAdapter
+                        .BuildProperties
+                        .GetPropertyValueAsync(ProjectBuildProperties.TargetFrameworkMoniker);
+
+                    var targetFrameworkMoniker = !string.IsNullOrWhiteSpace(targetFrameworkMonikerString)
+                        ? NuGetFramework.Parse(targetFrameworkMonikerString)
+                        : null;
+
+                    if (platformMinVersion != null && jsonTargetFramework.Version != platformMinVersion && FrameworkConstants.CommonFrameworks.NetCore50 == targetFrameworkMoniker)
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectJsonNuGetProject.cs:87]: Did you confirmed that all c# UAP based projects will have netcore50 as TFM? like how about PCL (with uap, desktop, netstandard)? Just confirm

**Review by jainaashish — CHANGES_REQUESTED** [review]
Please add tests to cover this change at NuGet side so that we dont regress it again.

**Review by emgarten — APPROVED** [review]
👍 on tests, then :shipit:

**Comment by rohit21agrawal** [comment]
@emgarten @jainaashish added test

**Review by jainaashish — APPROVED** [review]
Looks good

**Pull request merged**